scipy/signal/spectral.py

```diff
@@ -552,7 +561,9 @@ def csd(x, y, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
 
     # Average over windows.
     if len(Pxy.shape) >= 2 and Pxy.size > 0:
-        if Pxy.shape[-1] > 1:
+        if Pxy.shape[-1] > 1 and average == 'median':
+            Pxy = np.median(Pxy, axis=-1) / _median_bias(Pxy.shape[-1])
+        elif Pxy.shape[-1] > 1:
             Pxy = Pxy.mean(axis=-1)
```

Comment:
This would read more cleanly to me as:
```
if Pxy.shape[-1] > 1:
    if average == 'median':
        ...
    elif average == 'mean':
        ...
    else:
        raise ValueError('average must be "median" or "mean", got %s' % (average,))
```

Reply:
Not addressed